src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -731,20 +739,34 @@ abs_client::delete_object(
                  }),
                  key)
           .then([&name, &key](const ret_t& result) {
-              // ABS returns a 404 for attempts to delete a blob that doesn't
-              // exist. The remote doesn't expect this, so we map 404s to a
-              // successful response.
-              if (!result && result.error() == error_outcome::key_not_found) {
-                  vlog(
-                    abs_log.debug,
-                    "Object to be deleted was not found in cloud storage: "
-                    "object={}, bucket={}. Ignoring ...",
-                    name,
-                    key);
-                  return ss::make_ready_future<ret_t>(no_response{});
-              } else {
-                  return ss::make_ready_future<ret_t>(result);
+              if (!result) {
+                  if (result.error() == error_outcome::key_not_found) {
+                      // ABS returns a 404 for attempts to delete a blob that
+                      // doesn't exist. The remote doesn't expect this, so we
+                      // map 404s to a successful response.
+                      vlog(
+                        abs_log.debug,
+                        "Object to be deleted was not found in cloud storage: "
+                        "object={}, bucket={}. Ignoring ...",
+                        name,
+                        key);
+                      return ss::make_ready_future<ret_t>(no_response{});
+                  } else if (
+                    result.error() == error_outcome::operation_not_supported) {
+                      // ABS does not allow for deletion of directories when HNS
+                      // is disabled. The "folder" is "removed" when all blobs
+                      // inside of it are deleted. Map this to a successful
+                      // response.
+                      vlog(
```

Comment:
maybe it should be logged on 'info' or 'warn' level instead

Reply:
In the case of a mis-match between HNS  being enabled in `redpanda` and in Blob Storage, I would expect to see this log pop up a lot. I think keeping it at the `debug` level might be ideal.

Reply:
nit: frequency can be reduced with `ss::logger::rate_limit`